README.md

```diff
@@ -278,117 +166,129 @@ so you won't get Windows-specific functionality.
+
+     * to add to `~/.profile`:
+       ``` bash
+       echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+       echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+       echo 'eval "$(pyenv init -)"' >> ~/.profile
+       ```
+     * to add to `~/.bash_profile`:
+       ```bash
+       echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bash_profile
+       echo '[[ -d $PYENV_ROOT/bin ]] && export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bash_profile
+       echo 'eval "$(pyenv init -)"' >> ~/.bash_profile
+       ```
 
    **Bash warning**: There are some systems where the `BASH_ENV` variable is configured
    to point to `.bashrc`. On such systems, you should almost certainly put the
    `eval "$(pyenv init -)"` line into `.bash_profile`, and **not** into `.bashrc`. Otherwise, you
    may observe strange behaviour, such as `pyenv` getting into an infinite loop.
    See [#264](https://github.com/pyenv/pyenv/issues/264) for details.
-
-   **Proxy note**: If you use a proxy, export `http_proxy` and `https_proxy`, too.
-
-### Restart your shell
+   
+   </details>
+   
+#### Zsh
+  
+  <details>
+  
+  ```zsh
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.zshrc
+    echo '[[ -d $PYENV_ROOT/bin ]] && export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.zshrc
+    echo 'eval "$(pyenv init -)"' >> ~/.zshrc
+  ```
+  
+  If you wish to get Pyenv in noninteractive login shells as well, also add the commands to `~/.zprofile` or `~/.zlogin`.
+  </details>
+  
+#### Fish
+  
+  <details>
+    
+  1. If you have Fish 3.2.0 or newer, execute this interactively:
+     ~~~ fish
+       set -Ux PYENV_ROOT $HOME/.pyenv
+       fish_add_path $PYENV_ROOT/bin
+     ~~~
+
+  2. Otherwise, execute the snippet below:
+     ~~~ fish
+       set -Ux PYENV_ROOT $HOME/.pyenv
+       set -U fish_user_paths $PYENV_ROOT/bin $fish_user_paths
+     ~~~
+
+  3. Now, add this to `~/.config/fish/config.fish`:
+     ~~~ fish
+       pyenv init - | source
+     ~~~
+  </details>
+
+### C. Restart your shell
+----
 
   for the `PATH` changes to take effect.
 
   ```sh
   exec "$SHELL"
   ```
 
-### Install Python build dependencies
+### D. Install Python build dependencies
+----
 
   [**Install Python build dependencies**](https://github.com/pyenv/pyenv/wiki#suggested-build-environment)
   before attempting to install a new Python version.
 
   You can now begin using Pyenv.
 
+### E. Upgrade Notes
+----
+
+**if you have just upgraded to pyenv version >= 2.3.0**
+
+The startup logic and instructions have been updated for simplicity in 2.3.0.
+The previous, more complicated configuration scheme for 2.0.0-2.2.5 still works.
+
+* Define environment variable `PYENV_ROOT` to point to the path where
+  Pyenv will store its data. `$HOME/.pyenv` is the default.
+  If you installed Pyenv via Git checkout, we recommend
+  to set it to the same location as where you cloned it.
+* Add the `pyenv` executable to your `PATH` if it's not already there
```

Comment:
IMO This can be compacted, too (and probably deleted altogether after some time). Currently, it takes disproportionate space and visual weight for how situational it is.